src/Roslyn.Diagnostics.Analyzers/UnitTests/CSharpAvoidOptSuffixForNullableEnableCodeTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Roslyn.Diagnostics.CSharp.Analyzers.CSharpAvoidOptSuffixForNullableEnableCode,
+    Roslyn.Diagnostics.CSharp.Analyzers.CSharpAvoidOptSuffixForNullableEnableCodeCodeFixProvider>;
+
+namespace Roslyn.Diagnostics.Analyzers.UnitTests
+{
+    public class CSharpAvoidOptSuffixForNullableEnableCodeTests
+    {
+        [Fact]
+        public async Task RS0046_CSharp8_NullableEnabledCode_Diagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp8,
+                TestCode = @"
+#nullable enable
+
+public class Class1
+{
+    private Class1? [|_instanceOpt|], [|otherInstanceOpt|];
+
+    public void Method1(string? [|sOpt|])
+    {
+        string? [|localOpt|], [|otherLocalOpt|];
+    }
+}",
+                FixedCode = @"
+#nullable enable
+
+public class Class1
+{
+    private Class1? _instanceOpt, [|otherInstanceOpt|];
+
+    public void Method1(string? [|sOpt|])
+    {
+        string? [|localOpt|], [|otherLocalOpt|];
+    }
+}",
+                BatchFixedCode = @"
+#nullable enable
+
+public class Class1
+{
+    private Class1? _instance, otherInstance;
+
+    public void Method1(string? s)
+    {
+        string? other, otherLocalOpt;
+    }
+}",
```

Comment:
This isn't working and I am not sure what's going on. Breakpoints on the codefix provider are never reached.

Reply:
@sharwell?

Reply:
What's the test failure message? If it fails during diagnostic validation, the code fixes will not have run yet.

Reply:
Failure message:

```
  Microsoft.CodeAnalysis.Testing.Verifiers.EqualWithMessageException : Context: Diagnostics of fixed state
    Mismatch between number of diagnostics returned, expected "0" actual "1"
    
    Diagnostics:
    // /0/Test0.cs(10,24): warning RS0046: Avoid the 'Opt' suffix in a nullable-enabled code
    VerifyCS.Diagnostic().WithSpan(10, 24, 10, 37),
```

Reply:
My fixed code is invalid, I forgot to rename the variables!!! On previous versions of the test sdk the messages were a lot better we used to have the full diff displayed which was making comparison easier.